File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/inscription/DictionaryAdminService.java
package ma.zs.zyn.service.facade.admin.inscription;

import java.util.List;
import ma.zs.zyn.bean.core.inscription.Dictionary;
import ma.zs.zyn.dao.criteria.core.inscription.DictionaryCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface DictionaryAdminService extends  IService<Dictionary,DictionaryCriteria>  {

    List<Dictionary> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantRef(String ref);




}

File: frontend-bqckup/src/app/shared/service/admin/inscription/DictionaryAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {DictionaryDto} from '../../../model/inscription/Dictionary.model';
import {DictionaryCriteria} from '../../../criteria/inscription/DictionaryCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class DictionaryAdminService extends AbstractService<DictionaryDto, DictionaryCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/dictionary/';
    }

    public constrcutDto(): DictionaryDto {
        return new DictionaryDto();
    }

    public constrcutCriteria(): DictionaryCriteria {
        return new DictionaryCriteria();
    }
}
